docs/retrieve.txt

```diff
@@ -0,0 +1,248 @@
+
+For instructions on creating these classes, see the Quick Start.
+
+.. _laravel-find-all:
+
+Find All Documents
+------------------
+
+To retrieve all documents in a collection, use the ``all()`` method.
+
+To retrieve all documents that match a set of criteria, use the ``where()`` method.
+Pass the field name and value for which you are querying as parameters to ``where()``.
+
+You can also pass a comparison operator as the ``where()`` method's second parameter,
+such as the ``>`` operator. If you do not include a comparison operator parameter,
+the find operation queries for documents whose field values are equal to the specified
+value.
+
+To apply multiple sets of criteria to the find operation, you can chain a series
+of ``where()`` methods together.
+
+Find All Documents in a Collection Example
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The following example retrieves all documents in the ``movies`` collection:
+
+.. io-code-block::
+   :copyable: true
+
+   .. input::
+      :language: php
+
+      $movies = Movie::all();
+
+   .. output::
+      :language: none
+      :visible: false
+
+      // Results are truncated
+
+      ...
+
+      Title: Dracula
+      Year: 1931
+      Runtime: 85
+      IMDB Rating: 7.6
+      IMDB Votes: 30184
+      Plot: The ancient vampire Count Dracula arrives in England and begins to prey
+      upon the virtuous young Mina.
+
+      ...
+
+Find All Matching Documents Example
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This example chains two ``where()`` methods together to retrieve documents that meet
+the following criteria:
+
+- ``year`` field has a value of ``2010``
+- ``imdb.rating`` nested field has a value greater than ``8.5``
+
+.. io-code-block::
+   :copyable: true
+
+   .. input::
+      :language: php
+
+      $movies =
+         Movie::where('year', 2010)
+            ->where('imdb.rating', '>', 8.5)
+            ->get();
+
+   .. output::
+      :language: none
+      :visible: false
+
+      Title: Inception
+      Year: 2010
+      Runtime: 148
+      IMDB Rating: 8.8
+      IMDB Votes: 1294646
+      Plot: A thief who steals corporate secrets through use of dream-sharing
+      technology is given the inverse task of planting an idea into the mind of a CEO.
+
+      Title: Senna
+      Year: 2010
+      Runtime: 106
+      IMDB Rating: 8.6
+      IMDB Votes: 41904
+      Plot: A documentary on Brazilian Formula One racing driver Ayrton Senna, who won the
+      F1 world championship three times before his death at age 34.
+
+.. _laravel-find-one:
+
+Find One Document
+-----------------
+
+To retrieve one document that matches a set of criteria, use the ``where()`` method
+followed by the ``first()`` method. 
+
+For more information about the ``where()`` method and its parameters, see the
+:ref:`laravel-find-all` section of this guide.
+
+Find One Example
+~~~~~~~~~~~~~~~~
+
+The following example returns one document in which the value of the ``title`` field
+is ``"The Parent Trap"``:
```

Comment:
S: could you use an example here for a query that would definitely match more than one document to show that the method matches only the first one?

Reply:
I used "The Parent Trap" because there are two matching movie documents (the 1961 and 1998 versions) - would it be better to have a query that's even more broad / matches more documents?

Reply:
agreed!